dbt_platform_helper/domain/maintenance_page.py

```diff
@@ -275,13 +278,35 @@ def __clean_up_maintenance_page_rules(
             listener_arn
         )
 
-        deletes = {}
+        # keep track of rules deleted
+        deletes = {"MaintenancePage": 0}
         for name in ["MaintenancePage", "AllowedIps", "BypassIpFilter", "AllowedSourceIps"]:
-            deleted = self.load_balancer.delete_listener_rule_by_tags(tag_descriptions, name)
-            deletes[name] = bool(deleted)
-            if fail_when_not_deleted and name == "MaintenancePage" and not deleted:
+            deleted_list = self.load_balancer.delete_listener_rule_by_tags(tag_descriptions, name)
+
+            # track the rules deleted grouped by service
+            for deleted_rule in deleted_list:
+                tags = {t["Key"]: t["Value"] for t in deleted_rule["Tags"]}
+                if tags.get("service"):
```

Comment:
More pythonic to do this I think:
```suggestion
                if "service" in tags:
```